### PR TITLE
Daemon toolchain renames

### PR DESCRIPTION
**Pull request by ljacomet**
* Rename `gradle-build.properties` to `gradle-daemon-jvm.properties`
* Rename properties in `gradle-daemon-jvm.properties` 

@big-guy @vmadalin I am sorry I did not pay more attention during the spec process. But after trying out the feature a bit and showing it around, I feel strongly about a rename. I believe that the names I propose here are more focused and clarify that this introduced properties file has a clear purpose. It should also feel familiar to users configuring the wrapper.

* Why not `gradle-build-jvm.properties`?
  * I played with it but realised while looking for name and code references that we already have search hits on `gradle-build` but none on `gradle-daemon`
* Why the `jvm` part?
  * Again, to clarify purpose. This is not a bag for all things daemon, it is only for specifying the JVM to run it. Again making the parallel with the focus of the `gradle-wrapper.properties` file.
* Why the camelCase for property names and the removal of the `daemon.jvm` prefix?
  * Again, focus: these properties can only exist in a file that already has the `daemon-jvm` part in its name, no need to repeat it. And camel case makes them similar to wrapper properties. And that sends me back to the first point about the name of the file: get inspired by how we named properties.

Let me know if I missed anything around the rename.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test this

**Comment by ljacomet** [comment]
@bot-gradle test this

AFAICT failures are caused by an environment problem

File: platforms/core-runtime/build-configuration/src/integTest/groovy/org/gradle/interal/buildconfiguration/tasks/UpdateDaemonJvmIntegrationTest.groovy [hunk 1]
```diff
@@ -21,13 +21,13 @@ import org.gradle.api.JavaVersion
...
-import org.gradle.internal.buildconfiguration.BuildPropertiesDefaults
-import org.gradle.internal.buildconfiguration.fixture.BuildPropertiesFixture
+import org.gradle.internal.buildconfiguration.DaemonJVMPropertiesDefaults
+import org.gradle.internal.buildconfiguration.fixture.DaemonJVMPropertiesFixture
 import org.gradle.internal.jvm.Jvm
 import org.gradle.test.precondition.Requires
 import org.gradle.test.preconditions.IntegTestPreconditions
 
-class UpdateDaemonJvmIntegrationTest extends AbstractIntegrationSpec implements BuildPropertiesFixture {
+class UpdateDaemonJvmIntegrationTest extends AbstractIntegrationSpec implements DaemonJVMPropertiesFixture {
```
Comment (vmadalin) [inline platforms/core-runtime/build-configuration/src/integTest/groovy/org/gradle/interal/buildconfiguration/tasks/UpdateDaemonJvmIntegrationTest.groovy:30]: First this file should be considered to be named as `UpdateDaemonJVMIntegrationTest` keeping aligned with the renaming of other classes which uses the JVM with capital letters. This also applies to `UpdateDaemonJvm.java` and `UpdateDaemonJvmModifierTest.groovy`

However, this change goes against the naming already used for the existing classes. IMHO we should consider keeping everything as `Jvm` for consistency reasons 

Screenshot 2024-04-24 at 15 04 14

**Comment by vmadalin** [comment]
Thank you @ljacomet for those changes. In general, I would say that we had a long discussion about the following:
 - File location considering `<project>/gradle/...`, `<project>/gradle/daemon/...`, `<project>/gradle/build/...`, `<project>/gradle/buildEnvironment/...`
 - Camel case vs dot case properties format but also naming like `jvm.toolchain.version`, `daemon.jvm.toolchain.version`, `toolchain.version`
 - File name considering `gradle-build.properties`, `gradle-daemon.properties`, `gradle-init.properties`, `gradle-startup.properties`

There weren't any strong arguments, a part of reusing the same file to expose new properties for example to configure the build cache [mentioned](https://docs.google.com/document/d/1p0QohF1TAZOVWh9yCgsp97H6uDRAjHZeLNmD4dT1IGc/edit?disco=AAAA_U1V_LE) by Lorant. Something that potentially `gradle-daemon-jvm.properties` naming may restrict from being able to reuse the same file. However, I don't have all the details so probably will be great to double check with him.

A side of that, IMO make completely sense what you described which at the end of the day should make it easier for the user to understand the purpose of it, even though we don't expect users to manually modify the file since they should use the `updateDaemonJvm` task considering that this task will also populate the toolchains download urls in the future, using this just as a static/generated file.

File: platforms/core-runtime/build-configuration/src/integTest/groovy/org/gradle/interal/buildconfiguration/tasks/UpdateDaemonJvmIntegrationTest.groovy (hunk 1 above)
Comment (ljacomet) [reply platforms/core-runtime/build-configuration/src/integTest/groovy/org/gradle/interal/buildconfiguration/tasks/UpdateDaemonJvmIntegrationTest.groovy:30]: That's a very good point and I hesitated on it, without finding a way to answer my hesitation. Thanks for pointing these out!

Comment (ljacomet) [reply platforms/core-runtime/build-configuration/src/integTest/groovy/org/gradle/interal/buildconfiguration/tasks/UpdateDaemonJvmIntegrationTest.groovy:30]: Aligned to use `Jvm` everywhere

**Comment by ljacomet** [comment]
@bot-gradle test this

**Comment by ljacomet** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle merge

**Pull request merged**